src/Map.cc

```diff
@@ -359,7 +359,11 @@ void Map::SetLastMapChange(int currentChangeId)
 void Map::PreSave(std::set<GeometricCamera*> &spCams)
 {
     int nMPWithoutObs = 0;
-    for(MapPoint* pMPi : mspMapPoints)
+
+    std::set<MapPoint*> tmp_mspMapPoints1;
```

Comment:
these were pretty much the issue as to why slam was failing when trying to save a map, even at shutdown. some notes & where I pulled the code from on this error can be found [here](https://github.com/UZ-SLAMLab/ORB_SLAM3/issues/443#issuecomment-1097759891)